agimus_controller/agimus_controller/trajectory.py

```diff
@@ -37,14 +140,64 @@ class WeightedTrajectoryPoint:
     point: TrajectoryPoint
     weight: TrajectoryPointWeights
 
+    def __eq__(self, other):
+        if not isinstance(other, WeightedTrajectoryPoint):
+            return False
+
+        # Compare the 'point' and 'weight' attributes
+        if self.point != other.point:
+            return False
+
+        if self.weight != other.weight:
+            return False
 
-class TrajectoryBuffer(deque):
+        return True
+
+
+class TrajectoryBuffer(object):
     """List of variable size in which the HPP trajectory nodes will be."""
 
-    def clear_past(self, current_time_ns):
-        while self and self[0].point.time_ns < current_time_ns:
-            self.popleft()
+    def __init__(self, dt_factor_n_seq: list[tuple[int, int]]):
+        self._buffer = []
+        self.dt_factor_n_seq = deepcopy(dt_factor_n_seq)
+        self.horizon_indexes = self.compute_horizon_indexes(self.dt_factor_n_seq)
+
+    def append(self, item):
+        self._buffer.append(item)
+
+    def pop(self, index=-1):
+        return self._buffer.pop(index)
+
+    def clear_past(self):
+        if self._buffer:
```

Comment:
this would be clearer 
```suggestion
        if self._buffer != []:
```

Reply:
It's pythonic to test if a list is empty with a simple if.